prow/kube/client_test.go

```diff
@@ -230,6 +232,58 @@ func TestGetPod(t *testing.T) {
 	}
 }
 
+func TestGetLogTail(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			t.Errorf("Bad method: %s", r.Method)
+		}
+		if r.URL.Path != "/api/v1/namespaces/ns/pods/testpod/log" {
+			t.Errorf("Bad request path: %s", r.URL.Path)
+		}
+		limitQueryParam := r.URL.Query().Get("limitBytes")
+		byteLim, err := strconv.ParseInt(limitQueryParam, 10, 64)
+		if err != nil {
+			t.Fatalf("Invalid byte limit: %s, must be integer value", limitQueryParam)
+		}
+		var log []byte
+		for i := 0; i < 5; i++ {
+			log = append(log, []byte("What do you call recorded dolphin conversations?\npod logs")...)
+		}
+		logLen := int64(len(log))
+		if byteLim >= logLen {
+			fmt.Fprint(w, string(log))
+		} else {
+			fmt.Fprint(w, string(log[logLen-byteLim:]))
+		}
+	}))
+	defer ts.Close()
+	c := getClient(ts.URL)
+	testCases := []struct {
+		name     string
+		bytes    int64
+		expected []byte
+	}{
+		{
+			name:     "Get last 15 bytes of pod log",
```

Comment:
What happens when you request 100000 and there are only 100 bytes?


If you only need one testcase then don't both with the `range testCases` pattern

Reply:
addressed in latest commit